jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsContextParamHandler.java

```diff
@@ -0,0 +1,271 @@
+import org.jboss.as.controller.operations.validation.ParameterValidator;
+import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.as.jaxrs.logging.JaxrsLogger;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+
+public class JaxrsContextParamHandler {
+
+   // Attributes in alphabetical order
+   public static final String JAX_RS_2_0_REQUEST_MATCHING = "jaxrs.2.0.request.matching";
+   public static final String RESTEASY_ADD_CHARSET = "resteasy.add.charset";
+   public static final String RESTEASY_BUFFER_EXCEPTION_ENTITY = "resteasy.buffer.exception.entity";
+   public static final String RESTEASY_DISABLE_DTDS = "resteasy.document.secure.disableDTDs";
+   public static final String RESTEASY_DISABLE_HTML_SANITIZER = "resteasy.disable.html.sanitizer";
+   public static final String RESTEASY_DISABLE_PROVIDERS = "resteasy.disable.providers";
+   public static final String RESTEASY_EXPAND_ENTITY_REFERENCES = "resteasy.document.expand.entity.references";
+   public static final String RESTEASY_GZIP_MAX_INPUT = "resteasy.gzip.max.input";
+   public static final String RESTEASY_JNDI_RESOURCES = "resteasy.jndi.resources";
+   public static final String RESTEASY_LANGUAGE_MAPPINGS = "resteasy.language.mappings";
+   public static final String RESTEASY_MEDIA_TYPE_MAPPINGS = "resteasy.media.type.mappings";
+   public static final String RESTEASY_MEDIA_TYPE_PARAM_MAPPING = "resteasy.media.type.param.mapping";
+   public static final String RESTEASY_PROVIDERS = "resteasy.providers";
+   public static final String RESTEASY_RESOURCES = "resteasy.resources";
+   public static final String RESTEASY_RFC7232_PRECONDITIONS = "resteasy.rfc7232preconditions";
+   public static final String RESTEASY_ROLE_BASED_SECURITY = "resteasy.role.based.security";
+   public static final String RESTEASY_SECURE_PROCESSING_FEATURE = "resteasy.document.secure.processing.feature";
+   public static final String RESTEASY_SECURE_RANDOM_MAX_USE = "resteasy.secure.random.max.use";
+   public static final String RESTEASY_USE_BUILTIN_PROVIDERS = "resteasy.use.builtin.providers";
+   public static final String RESTEASY_USE_CONTAINER_FORM_PARAMS = "resteasy.use.container.form.params";
+   public static final String RESTEASY_WIDER_REQUEST_MATCHING = "resteasy.wider.request.matching";
+
+   private static final String JAVA_ID = "[a-zA-Z_$][a-zA-Z_$0-9]*(\\.[a-zA-Z_$][a-zA-Z_$0-9]*)*";
+   private static final String JNDI_ID = JAVA_ID + ":" + JAVA_ID + "(/" + JAVA_ID + ")*";
+   private static final Pattern PATTERN_LIST_ELEMENT = Pattern.compile(JAVA_ID + "|" + JNDI_ID);
+   private static final Pattern PATTERN_PROPERTY_LIST_ELEMENT = Pattern.compile(".+:.+(,.+:.+)*");
+   private static final String NO_DEFAULT = "noDefault";
+
+   private static final Map<String, ModelNode> contextParams = new TreeMap<String, ModelNode>();
+
+   private static final Map<String, Boolean> contextParametersBoolean = new HashMap<String, Boolean>();
```

Comment:
I noticed when trying to edit your code that you are using 3 spaces for indentation. The WildFly code style is 4 spaces. I've not checked everywhere, but I presume it will apply to all the code you have done due to your IDE settings?